geopandas/tools/tests/test_sjoin.py

```diff
@@ -488,3 +506,189 @@ def test_sjoin_inner(self):
             self.cities, countries, how="inner", op="intersects"
         )
         assert cities_with_country.shape == (172, 4)
+
+
+@pytest.mark.skipif(
+    TEST_NEAREST,
+    reason=(
+        f"PyGEOS >= {MIN_PYGEOS_VERSION_FOR_NEAREST}"
+        " must be installed and activated via the geopandas.compat module to"
+        " test sjoin_nearest"
+    ),
+)
+def test_no_nearest_all():
+    df1 = geopandas.GeoDataFrame({"geometry": []})
+    df2 = geopandas.GeoDataFrame({"geometry": []})
+    with pytest.raises(
+        AttributeError, match="Expected the spatial index to implement `nearest_all`"
+    ):
+        sjoin_nearest(df1, df2)
+
+
+@pytest.mark.skipif(
+    not TEST_NEAREST,
+    reason=(
+        f"PyGEOS >= {MIN_PYGEOS_VERSION_FOR_NEAREST}"
+        " must be installed and activated via the geopandas.compat module to"
+        " test sjoin_nearest"
+    ),
+)
+class TestNearest:
+    @pytest.mark.skipif(TEST_NEAREST)
```

Comment:
Why is this here?

Reply:
Weird, I can't think of why I would have put that there. I do remember some strangeness like setup_method running even when the class itself was skipped, but if that was the case it should be negated...

Reply:
I have some vague recollection of adding it for a reason, but I can't reproduce (if I remove it things still run fine with and without pygeos) and it doesn't make sense to me now, so I'm going to remove and see if all is okay in CI.